scripts/west_commands/build.py

```diff
@@ -127,10 +143,29 @@ def do_run(self, args, remainder):
                                                        self.args.cmake_opts))
         self._sanity_precheck()
         self._setup_build_dir()
+
+        if args.pristine is not None:
+            pristine = args.pristine
+        else:
+            # Load the pristine={auto, always, never} configuration value
+            pristine = config.get('build', 'pristine', fallback='never')
+            if pristine not in ['auto', 'always', 'never']:
+                log.wrn('treating unknown build.pristine value "{}" as "never"'.
+                        format(pristine))
+                pristine = 'never'
+        self.auto_pristine = True if pristine == 'auto' else False
```

Comment:
```suggestion
        self.auto_pristine = pristine == 'auto'
```

Thanks for renaming. This is also a bit redundant, since `==` will always evaluate to True or False.

You could also do `self.auto_pristine = (pristine == 'auto')` if that's more clear.

Reply:
Fixed! For some reason this did not work when I first wrote it and switched to the longer, explicit version. But now I re-tried and worked fine so I agree this is the correct way to do this.

Reply:
Great, looking very nice, thanks!